README.md

```diff
@@ -1,145 +1,308 @@
-
-   1. `GOOGLE_APPLICATION_CREDENTIALS` (again)
-   2. gcloud _application default_ credentials (set from ` gcloud auth application-default login`)
-   3. appengine.AccessToken (for App Engine Go < =1.9)
-   4. GCE/GAE metadata credentials
-
-When the proxy authenticates under the default service account of the
-Compute Engine VM it is running on the VM must have at least the
-sqlservice.admin API scope ("https://www.googleapis.com/auth/sqlservice.admin")
-and the associated project must have the SQL Admin API
-enabled.  The default service account must also have at least WRITER/EDITOR
-priviledges to any projects of target SQL instances.
-
-Specifying the `-credential_file` flag allows use of the proxy outside of
-Google's cloud. Simply [create a new service
+3. [Application Default Credentials](https://cloud.google.com/docs/authentication/production)
+
+Any account connecting to a Cloud SQL database will need one of the following
+IAM roles:
+
+- Cloud SQL Client (preferred)
+- Cloud SQL Editor
+- Cloud SQL Admin
+
+Or you can manually assign the following IAM permissions:
+
+- cloudsql.instances.connect
+- cloudsql.instances.get
+
+See [Roles and Permissions in Cloud
+SQL](https://cloud.google.com/sql/docs/mysql/roles-and-permissions) for details.
+
+When the proxy authenticates under the default service account of the Compute
+Engine VM it is running on the VM must have at least the sqlservice.admin API
+scope ("https://www.googleapis.com/auth/sqlservice.admin") and the associated
+project must have the SQL Admin API enabled. The default service account must
+also have at least WRITER/EDITOR priviledges to any projects of target SQL
+instances.
+
+* `-credential_file`
+
+Allows use of the proxy outside of Google's cloud. Simply [create a new service
 account](https://cloud.google.com/sql/docs/mysql/sql-proxy#create-service-account),
 download the associated JSON file, and set `-credential_file` to the path of the
-JSON file. You can also set the GOOGLE_APPLICATION_CREDENTIALS environment variable
-instead of passing this flag.
+JSON file. You can also set the `GOOGLE_APPLICATION_CREDENTIALS` environment
+variable instead of passing this flag.
 
-## Example invocations:
+* `-token`
 
-    ./cloud_sql_proxy -dir=/cloudsql -instances=my-project:us-central1:sql-inst &
-    mysql -u root -S /cloudsql/my-project:us-central1:sql-inst
+When set, the proxy uses this Bearer token for authorization.
 
-    # To retrieve instances from a custom metadata value (only when running on GCE)
-    ./cloud_sql_proxy -dir=/cloudsql -instances_metadata instance/attributes/<custom-metadata-key> &
-    mysql -u root -S /cloudsql/my-project:us-central1:sql-inst
+* `-enable_iam_login`
 
-    # For -fuse you do not need to specify instance names ahead of time:
-    ./cloud_sql_proxy -dir=/cloudsql -fuse &
-    mysql -u root -S /cloudsql/my-project:us-central1:sql-inst
+Enables the proxy to use Cloud SQL IAM database authentication. This will cause
```

Comment:
Let's add a link to this page: https://cloud.google.com/sql/docs/postgres/authentication

Reply:
Done.